packages/video_player_videohole/tizen/src/messages.h

```diff
@@ -0,0 +1,312 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+// Autogenerated from Pigeon (v3.1.4), do not edit directly.
+// See also: https://pub.dev/packages/pigeon
+
+#ifndef PIGEON_MESSAGES_H_
+#define PIGEON_MESSAGES_H_
+#include <flutter/basic_message_channel.h>
+#include <flutter/binary_messenger.h>
+#include <flutter/encodable_value.h>
+#include <flutter/standard_message_codec.h>
+
+#include <map>
+#include <optional>
+#include <string>
+
+/* Generated class from Pigeon. */
+
+class FlutterError {
+ public:
+  FlutterError();
+  FlutterError(const std::string& arg_code) : code(arg_code){};
+  FlutterError(const std::string& arg_code, const std::string& arg_message)
+      : code(arg_code), message(arg_message){};
+  FlutterError(const std::string& arg_code, const std::string& arg_message,
+               const flutter::EncodableValue& arg_details)
+      : code(arg_code), message(arg_message), details(arg_details){};
+  std::string code;
+  std::string message;
+  flutter::EncodableValue details;
+};
+template <class T>
+class ErrorOr {
+  std::variant<std::unique_ptr<T>, T, FlutterError> v;
+
+ public:
+  ErrorOr(const T& rhs) { new (&v) T(rhs); }
+  ErrorOr(const FlutterError& rhs) { new (&v) FlutterError(rhs); }
+  static ErrorOr<std::unique_ptr<T>> MakeWithUniquePtr(std::unique_ptr<T> rhs) {
+    ErrorOr<std::unique_ptr<T>> ret = ErrorOr<std::unique_ptr<T>>();
+    ret.v = std::move(rhs);
+    return ret;
+  }
+  bool hasError() const { return std::holds_alternative<FlutterError>(v); }
+  const T& value() const { return std::get<T>(v); };
+  const FlutterError& error() const { return std::get<FlutterError>(v); };
+
+ private:
+  ErrorOr() = default;
+  template <typename U>
+  friend class ErrorOr;
+};
+
+/* Generated class from Pigeon that represents data sent in messages. */
+class TextureMessage {
```

Comment:
Unlike `video_player_tizen`, this plugin does not use textures, because it is based on the hole.

and the `player` is definitely overlaying the output.
```c++
    if (player_set_ecore_wl_display) {
      ret = player_set_ecore_wl_display(
          player_, PLAYER_DISPLAY_TYPE_OVERLAY,
          FlutterDesktopViewGetNativeHandle(view_ref), 0, 0, w, h);
    }
```

and you don't even use `flutter::TextureRegistrar`.

If so, my question is why did you use 'texture' in terms? also, in the code snippet below, player->Create() returns the ID of the player.

```c++
ErrorOr<std::unique_ptr<TextureMessage>> VideoPlayerTizenPlugin::Create(
    const CreateMessage &createMsg) {
  std::unique_ptr<VideoPlayer> player =
      std::make_unique<VideoPlayer>(registrar_ref_, createMsg);
  int64_t textureId = player->Create();
  players_[textureId] = std::move(player);
  std::unique_ptr<TextureMessage> texture_message =
      std::make_unique<TextureMessage>();
  texture_message->set_texture_id(textureId);
  return ErrorOr<TextureMessage>::MakeWithUniquePtr(std::move(texture_message));
}
```

Shouldn't the term `texture` remove throughout the implementation?

Reply:
If you added unnecessary code based on the `video_player_tizen`, please delete all of them.